pandas/core/dtypes/base.py

```diff
@@ -212,8 +209,7 @@ def __str__(self):
         return self.name
 
     @property
-    def type(self):
-        # type: () -> Type
+    def type(self) -> type:
```

Comment:
Was this for compatibility? Think `Type[Any]` would be preferable though not available until 3.5.2

Reply:
Not sure. According to the [typing docs](https://docs.python.org/3/library/typing.html), "`Type[Any]` is equivalent to `Type` which in turn is equivalent to `type`, which is the root of Python’s metaclass hierarchy," so it shouldn't matter. But mypy complains with it as-is. My vote is for `def type(self) -> Type`.

Python 3.5.0 implements PEP 484, so I think Type is available to us. What isn't available until 3.5.2?

Reply:
Right I think Type is best here - what error were you getting before?

Reply:
```
Performing static analysis using mypy
pandas/core/dtypes/base.py:212: error: Invalid type "pandas.core.dtypes.base.ExtensionDtype.type"
```
Pushing big-t Type now.

Reply:
Had a weird test failure, but the board is green on an empty commit.